client_test.go

```diff
@@ -49,6 +42,53 @@ func buildGoodResponse(data interface{}) ([]byte, error) {
 
 }
 
+func buildInternalServerErrorResponse(data interface{}) ([]byte, error) {
+	resp := &Response{
+		Status:		 "err",
+		Code:			 InternalServerErrorCode,
+		Host:			 "test",
+		Timestamp: 1234567890,
+		Message:	 "Internal Server Error",
+		Data:			 data,
+	}
+	return json.Marshal(resp)
+
+}
+
+// buildServer returns a server. Call Close when finished.
+func buildServer(code int, body []byte, a func(r *http.Request)) *httptest.Server {
+	return httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		a(r)
+		w.WriteHeader(code)
+		w.Header().Set("Content-Type", "application/json")
+		w.Write(body)
+	}))
+}
+
+func buildConcurrentServer(code int, t *testing.T, a func(r *http.Request)) *httptest.Server {
```

Comment:
instead of ops being a global atomic variable it probably would make more sense to declare it in here and then pass the current value into `a`.

Also be sure to run the tests with -race since you are adding multithreading code here.

Reply:
Yeah i agree ops shouldn't be global, I instantiate it in the test case instead and move the response generation logic (OK response vs. internal server error response) to the test case definition, i feel this makes more sense since we want to reproduce the retry logic in that particular test case